v2/tools/generator/internal/config/configuration.go

```diff
@@ -312,6 +312,125 @@ func getModulePathFromModFile(modFilePath string) (string, error) {
+		return err
+	}
+	if err := mergePrimitiveString(&config.TypeRegistrationOutputFile, other.TypeRegistrationOutputFile, "TypeRegistrationOutputFile"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.RootURL, other.RootURL, "RootURL"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveString(&config.SamplesPath, other.SamplesPath, "SamplesPath"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveEnum(&config.Pipeline, other.Pipeline, "Pipeline"); err != nil {
+		return err
+	}
+	if err := mergePrimitiveBool(&config.EmitDocFiles, other.EmitDocFiles, "EmitDocFiles"); err != nil {
+		return err
+	}
+
+	// Merge slice fields
+	config.AnyTypePackages = mergeSlice(config.AnyTypePackages, other.AnyTypePackages)
+	config.TypeFilters = mergeSlice(config.TypeFilters, other.TypeFilters)
+	config.TypeLoaderRenames = mergeSlice(config.TypeLoaderRenames, other.TypeLoaderRenames)
+	config.Transformers = mergeSlice(config.Transformers, other.Transformers)
+
+	// Merge complex fields
+	if err := config.Status.Merge(&other.Status); err != nil {
+		return err
+	}
+	
+	if other.SupportedResourcesReport != nil {
+		if config.SupportedResourcesReport == nil {
+			config.SupportedResourcesReport = NewSupportedResourcesReport(config)
+		}
+		if err := config.SupportedResourcesReport.Merge(other.SupportedResourcesReport); err != nil {
+			return err
+		}
+	}
+	
+	if other.ObjectModelConfiguration != nil {
+		if config.ObjectModelConfiguration == nil {
+			config.ObjectModelConfiguration = NewObjectModelConfiguration()
+		}
+		if err := config.ObjectModelConfiguration.Merge(other.ObjectModelConfiguration); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Helper functions for merging different types
+
+func mergePrimitiveString(base *string, other string, fieldName string) error {
+	if other == "" {
+		return nil // Nothing to merge
+	}
+	if *base == "" {
+		*base = other
+		return nil
+	}
+	if *base != other {
+		return fmt.Errorf("conflict in field %s: base value %q cannot be overwritten with %q", fieldName, *base, other)
+	}
+	return nil
+}
+
+func mergePrimitiveBool(base *bool, other bool, fieldName string) error {
+	if !other {
+		return nil // Nothing to merge (false is zero value)
+	}
+	if !*base {
+		*base = other
+		return nil
+	}
+	// Both are true - this is a conflict if they're both explicitly set
+	return fmt.Errorf("conflict in field %s: base value %v cannot be overwritten with %v", fieldName, *base, other)
+}
+
+func mergePrimitiveEnum[T comparable](base *T, other T, fieldName string) error {
+	var zero T
+	if other == zero {
+		return nil // Nothing to merge (zero value)
+	}
+	if *base == zero {
+		*base = other
+		return nil
+	}
+	if *base != other {
+		return fmt.Errorf("conflict in field %s: base value %v cannot be overwritten with %v", fieldName, *base, other)
+	}
+	return nil
+}
+
+func mergeSlice[T any](base []T, other []T) []T {
+	if len(other) == 0 {
```

Comment:
minor: This `len` check is useless - `mergeSlice` could just be `append(base, other...)` I think?